docs_src/design/adr/device-service/0021-Profile-Changes.md

```diff
@@ -0,0 +1,126 @@
+# Rules on Changes to a Device Profile
+
+## Status
+
+** Proposed **
+(as of 10/11/21)
+
+## Context
+
+While the device profile has always been the way to describe a device/sensor and template its communications to the rest of the system, over the course of EdgeX evolution there have been changes in what could change in a profile or its associations.  This document is meant to address the issue of change surrounding device profiles in EdgeX going forward.
+
+### Summary of Allowed changes
+
+A detailed look at the allowed changes to device profile and its device resources is listed below.  In general, here is what we are trying to allow/prevent via these rules:
+- Block the DELETE of a device profile when it is associated to any other EdgeX object (device, reading, event) **always**.
+- Allow changes to device profile or device resource fields when the field is inconsequential to the object's use (ex: description) or when it is not yet associated to another EdgeX object.  These changes are listed as `Anytime` or `Situational` accordingly in the table below.
+- Allow the addition of new profiles, device resources, device commands, etc. **always**.  Additions may be accomplished via POST or PATCH depending on the change and API.
+
+See [Proposed Design section](#proposed-design) section below for specific metadata API changes.
+
+While not explicitly covered in this ADR, a device is not allowed when it is associated to an event or reading.
```

Comment:
I'm not sure what you mean by this? Seems like something got left out here?

Reply:
Sorry @tonyespy - you are correct.  Cut and paste error that I am correcting.  should be "a device is not allowed to be deleted when it is associated to an event or reading"